app/pages/learn/01_tutorial/04_mastering-the-api/02_invoke/00_methodhandle.md

```diff
@@ -0,0 +1,628 @@
+that method handles cannot, which is listing the class members and inspecting their properties. Method handles, on the
+other hand, can be transformed and manipulated in a way that is not possible with Reflection API.
+
+When it comes to method invocation, there are differences related to access checking and security considerations. The
+Reflection API performs access checking against every caller, on every call, while for method handles, access is
+checked only during construction. This makes invocation through method handles faster than through reflection.
+However, certain precautions have to be taken so the method handle is not passed to the code where it shouldn't be
+accessible.
+
+<a id="unreflect">&nbsp;</a>
+## Conversion between Reflection API and method handles
+The `Lookup` object can be used to convert Reflection API objects to behaviorally equivalent method handles, which
+provide more direct and efficient access to the underlying class members.
+
+To create a method handle pointing to a given [`Method`](javadoc:Method) (given that the lookup class has permission to do so), we can
+use `unreflect`.
+
+Let's say we have a `test` method in our `Example` class which accepts a `String` argument and returns a `String`. Using
+the Reflection API, we can obtain a `Method` object:
+
+```java
+Method method = Example.class.getMethod("test", String.class);
+```
+
+With the help of the lookup object, we can [`unreflect`](javadoc:MethodHandles.Lookup.unreflect(Method)) the `Method`
+object to obtain a `MethodHandle`:
+
+```java
+MethodHandles.Lookup lookup = MethodHandles.lookup();
+MethodHandle methodHandle = lookup.unreflect(method);
+String result = (String) methodHandle.invoke("something");
+```
+
+Similarly, given a [`Field`](javadoc:Field) object, we can obtain getter and setter method handles:
+
+```java
+Field field = Example.class.getField("magic");
+MethodHandle setterMethodHandle = lookup.unreflectSetter(field);
+MethodHandle getterMethodHandle = lookup.unreflectGetter(field);
+setterMethodHandle.invoke("something");
+String result = (String) getterMethodHandle.invoke(); // result == "something"
+```
+
+Conversion from `MethodHandle` to a [`Member`](javadoc:Member) is also possible, with the condition that no transformations
+have been performed to the given `MethodHandle`.
+
+Let's say we have a method handle pointing directly to a method. We can use the
+[`MethodHandles.reflectAs`](javadoc:MethodHandles.reflectAs(Class,MethodHandle)) method to obtain the `Method` object:
+
+```java
+Method method = MethodHandles.reflectAs(Method.class, methodHandle);
+```
+
+It works similarly for the `Field` object:
+
+```java
+Field field = MethodHandles.reflectAs(Field.class, getterMethodHandle); // same result is achieved by reflecting `setterMethodHandle`
+```
+
+<a id="conclusion">&nbsp;</a>
+## Conclusion
+In this tutorial, we have looked into the method handle mechanism and learned how to efficiently use it. We now know,
```

Comment:
Can you please transform the conclusion to having several phrases (without the bullet list) having the advantages of using method handles? Also, below phrase confused me to understand what is easy or not 😁 

_Method handles make it easy to wrap the direct pointers to methods and fields into more complicated logic which
  includes argument and return value manipulations._